host/xtest/pkcs11_1000.c

```diff
@@ -3741,3 +3741,428 @@ static void xtest_pkcs11_test_1016(ADBG_Case_t *c)
 }
 ADBG_CASE_DEFINE(pkcs11, 1016, xtest_pkcs11_test_1016,
 		 "PKCS11: Random number generator tests");
+
+static CK_RV derive_sym_key(CK_SESSION_HANDLE session,
+			    CK_OBJECT_HANDLE parent_key,
+			    CK_MECHANISM_TYPE mechanism, size_t data_len,
+			    CK_OBJECT_HANDLE_PTR derv_key_hdl, size_t keylen,
+			    CK_OBJECT_CLASS keyclass, CK_KEY_TYPE keytype,
+			    CK_BBOOL sensitive, CK_BBOOL extble)
+{
+	CK_RV rv = CKR_GENERAL_ERROR;
+	uint8_t buffer[516] = { 0 };
```

Comment:
why 516?

Reply:
I will change it. When initially writing the test cases, I was assuming a generic secret key also which I realized was wrong. So , took buffer of size > 512. Will change ti to 48.